src/main/java/application/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,126 @@
+package application.service.impl;
+
+import application.dto.order.OrderRequestShippingAddressDto;
+import application.dto.order.OrderRequestStatusDto;
+import application.dto.order.OrderResponseDto;
+import application.exception.EntityNotFoundException;
+import application.mapper.OrderItemMapper;
+import application.mapper.OrderMapper;
+import application.model.CartItem;
+import application.model.Order;
+import application.model.OrderItem;
+import application.model.ShoppingCart;
+import application.repository.OrderRepository;
+import application.service.CartItemService;
+import application.service.OrderItemService;
+import application.service.OrderService;
+import application.service.ShoppingCartService;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private static final String FIND_ORDER_EXCEPTION = "Can't find order by id ";
+    private final OrderItemMapper orderItemMapper;
+    private final ShoppingCartService shoppingCartService;
+    private final CartItemService cartItemService;
+    private final OrderItemService orderItemService;
+    private final OrderRepository orderRepository;
+    private final OrderMapper orderMapper;
+
+    @Override
+    public OrderResponseDto createOrder(ShoppingCart shoppingCart,
+                                        OrderRequestShippingAddressDto dto) {
+        Order order = setAllFieldsOrder(shoppingCart, dto);
+        orderRepository.save(order);
+        Set<CartItem> cartItemSet = shoppingCart.getCartItemSet();
+        shoppingCartService.clearShoppingCart(shoppingCart);
+        deleteAllCartItems(cartItemSet);
+        OrderResponseDto responseDto = orderMapper.toResponseDto(order);
+        return setOrderItemsDtos(responseDto, order);
+    }
+
+    @Override
+    public List<OrderResponseDto> getListOrderResponseDtos(Long id) {
+        return findAllByUserId(id)
+                .stream()
+                .map(orderMapper::toResponseDto)
+                .map(c -> setOrderItemsDtos(c, findByOrderId(c.getId())
+                )).toList();
+
+    }
+
+    @Override
+    public void updateStatusOrder(Long id, OrderRequestStatusDto dto) {
+        Order order = findByOrderId(id);
+        order.setStatus(dto.getStatus());
+        orderRepository.save(order);
+    }
+
+    @Override
+    public List<Order> findAllByUserId(Long id) {
+        return orderRepository.findAllByUserId(id);
+    }
+
+    @Override
+    public Order findByOrderId(Long id) {
+        return orderRepository.findById(id)
+                .orElseThrow(() ->
+                        new EntityNotFoundException(FIND_ORDER_EXCEPTION + id));
+    }
+
+    private Order setAllFieldsOrder(ShoppingCart sc, OrderRequestShippingAddressDto dto) {

```

Comment:
Please analyse all your method names on grammar mistakes. In this context, more correct would be `setAllOrderFields`, but it's also not the best name for this method. Think about something like `prepareOrder` or similar.